packages/python-editor/src/KernelDropdown.tsx

```diff
@@ -16,128 +16,86 @@
-      .forEach(entry => delete specs.kernelspecs[entry[0]]);
-  };
-
-  /**
-   * Function: Creates drop down options with available python kernel specs.
-   */
-  private createOptionElems = (specs: KernelSpec.ISpecModels): void => {
-    let i = 0;
-    for (const name of Object.keys(specs.kernelspecs)) {
-      const spec = specs.kernelspecs[name];
-      const displayName = spec.display_name ? spec.display_name : name;
-      const elem = React.createElement(
-        'option',
-        { key: i++, value: name },
-        displayName
-      );
-      this.kernelOptionElems.push(elem);
-    }
-  };
-
-  /**
-   * Function: Handles kernel selection from dropdown options.
-   */
-  private handleSelection = (event: any): void => {
-    const selection: string = event.target.value;
-    this.updateKernel(selection);
-  };
-
-  render(): React.ReactElement {
-    return this.state.kernelSpecs
-      ? React.createElement(
-          HTMLSelect,
-          {
-            className: DROPDOWN_CLASS,
-            onChange: this.handleSelection.bind(this),
-            defaultValue: this.state.kernelSpecs.default
-          },
-          this.kernelOptionElems
-        )
-      : React.createElement('span', null, 'Fetching kernel specs...');
-  }
-}
+  const [selection, setSelection] = useState(initVal);
+
+  // Note: It's normally best to avoid using an imperative handle if possible.
+  // The better option would be to track state in the parent component and handle
+  // the change events there as well, but I know this isn't always possible
+  // alongside jupyter.
+  useImperativeHandle(select, () => ({
+    getSelection: (): string => selection
+  }));
+
+  const kernelOptions = !Object.keys(specs.kernelspecs).length ? (
+    <option key="no-kernel" value="no-kernel">
+      No Kernel
+    </option>
+  ) : (
+    Object.entries(specs.kernelspecs).map(([key, val]) => (
+      <option key={key} value={key}>
+        {val.display_name || key}
+      </option>
+    ))
+  );
+
+  return (
+    <select
+      className={KERNEL_SELECT_CLASS}
+      onChange={(e): void => setSelection(e.target.value)}
+      value={selection}
+    >
+      {kernelOptions}
+    </select>
+  );
+});
 
 /**
- * Class: A CellTypeSwitcher widget that renders the Dropdown component.
+ * Wrap the dropDown into a React Widget in order to insert it into a Lab Toolbar Widget
  */
 export class KernelDropdown extends ReactWidget {
-  private runner: PythonRunner;
-  private updateKernel: Function;
-
   /**
    * Construct a new CellTypeSwitcher widget.
    */
-  constructor(runner: PythonRunner, updateKernel: Function) {
+  constructor(
+    private specs: KernelSpec.ISpecModels,
+    private ref: RefObject<ISelect>
+  ) {
     super();
-    this.runner = runner;
-    this.updateKernel = updateKernel;
   }
 
   render(): React.ReactElement {
     return (
-      <DropDown {...{ runner: this.runner, updateKernel: this.updateKernel }} />
+      <div>
```

Comment:
any particular reason you're wrapping this in a `div`?

Reply:
You're right, that can be removed